v1.go

```diff
@@ -22,17 +23,20 @@ func initVersion1() *Header {
 }
 
 func parseVersion1(reader *bufio.Reader) (*Header, error) {
-	// Read until LF shows up, otherwise fail.
-	// At this point, can't be sure CR precedes LF which will be validated next.
-	line, err := reader.ReadString('\n')
-	if err != nil {
+	// The header must be terminated by CRLF and cannot be more than 107 bytes long.
+	buf, err := reader.Peek(107)
+	if err != nil && err != io.EOF {
```

Comment:
Per docs:
> If Peek returns fewer than n bytes, it also returns an error explaining why the read is short.
Is the error `io.EOF`? It's unclear from the docs and the code didn't immediately tell me it is.

Reply:
So there could be different errors - e.g. network read error. However, io.EOF is returned when there is less than 107 bytes available in the stream at that time, based on my tests. I will add some unit tests to cover various cases to validate this.